core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -227,9 +214,6 @@ public void repartition(final String topic, final int partitionsNumber) throws C
             if (!Boolean.TRUE.equals(areNewPartitionsAdded)) {
                 throw new TopicConfigException(String.format("Failed to repartition topic to %s", partitionsNumber));
             }
-            final Producer<byte[], byte[]> producer = kafkaFactory.takeProducer(topic);
-            kafkaFactory.terminateProducer(producer);
-            kafkaFactory.releaseProducer(producer);
```

Comment:
This code was here only to make re-partitioning tests pass.

The proper fix is not to use a producer when listing partitions.